core/src/main/java/org/apache/calcite/sql/SqlIntervalQualifier.java

```diff
@@ -746,7 +746,8 @@ private int[] evaluateIntervalLiteralAsQuarter(
       checkLeadFieldInRange(typeSystem, sign, quarter, TimeUnit.QUARTER, pos);
 
       // package values up for return
-      return fillIntervalValueArray(sign, ZERO, quarter);
+      final BigDecimal months = quarter.multiply(BigDecimal.valueOf(3));
```

Comment:
I don't think quarters should be convertible to seconds, they do not have a fixed length

Reply:
They are being converted to 3 months here, since QUARTER is treated as a kind of YEAR-MONTH interval.

Reply:
sorry, I misread the JIRA issue

Reply:
Ah, ok!